node/src/components/consensus/highway_core/state/tests.rs

```diff
@@ -459,13 +462,314 @@ fn validate_lnc_two_equivocators() -> Result<(), AddUnitError<TestContext>> {
+    // Dan            |        d0<---------+ |
+    //                |                      |
+    // Eric           +--+e0<----------------+
+    //
+    let weights_dan_eric = &[Weight(3), Weight(4), Weight(5), Weight(5), Weight(6)];
+    let mut state = State::new_test(weights_dan_eric, 0);
+    let mut rng = crate::new_rng();
+    let c0 = add_unit!(state, rng, CAROL, 0xC; N, N, N, N, N)?;
+    let c1 = add_unit!(state, rng, CAROL, 0xC1; N, N, c0, N, N)?;
+    let c1_prime = add_unit!(state, rng, CAROL, 0xC1B; N, N, c0, N, N)?;
+    let b0 = add_unit!(state, rng, BOB, 0xB; N, N, c1, N, N)?;
+    let d0 = add_unit!(state, rng, DAN, 0xD; N, N, c1_prime, N, N)?;
+    // Should not require endorsements b/c e0 sees Carol as correct.
+    let e0 = add_unit!(state, rng, ERIC, 0xE; N, N, c0, N, N)?;
+    let mut a0 = WireUnit {
+        panorama: panorama!(N, b0, F, d0, e0),
+        creator: ALICE,
+        instance_id: 1u64,
+        value: None,
+        seq_number: 0,
+        timestamp: 52.into(),
+        round_exp: 4u8,
+        endorsed: BTreeSet::new(),
+    };
+    assert_eq!(state.validate_lnc(&a0), Some(CAROL));
+    // We pick b0 to be endorsed.
+    a0.endorsed.insert(b0);
+    endorse!(state, rng, b0; ALICE, BOB, ERIC);
+    assert_eq!(state.validate_lnc(&a0), None);
+    Ok(())
+}
+
+#[test]
+#[ignore = "stubbed validate_lnc"]
+fn validate_lnc_transitive_endorsement() -> Result<(), AddUnitError<TestContext>> {
+    // Endorsements should be transitive to descendants.
+    // c1 doesn't have to be endorsed, it is enough that c0 is.
+    //
+    // Alice           a0<-----------+
+    //                 +             |
+    //          b0<----+             |
+    // Bob                           |
+    //                               |
+    //          b0'<---+             |
+    //                 +             |
+    // Carol           c0<---+c1<----+
+    //                               |
+    //                               |
+    // Dan                          d0
+
+    let weights_dan = &[Weight(3), Weight(4), Weight(5), Weight(5)];
+    let mut state = State::new_test(weights_dan, 0);
+    let mut rng = crate::new_rng();
+    let b0 = add_unit!(state, rng, BOB, 0xB; N, N, N, N)?;
+    let b0_prime = add_unit!(state, rng, BOB, 0xB1; N, N, N, N)?;
+    let a0 = add_unit!(state, rng, ALICE, 0xA; N, b0, N, N)?;
+    let c0 = add_unit!(state, rng, CAROL, 0xC; N, b0_prime, N, N)?;
+    let c1 = add_unit!(state, rng, CAROL, 0xC1; N, b0_prime, c0, N)?;
+    let mut d0 = WireUnit {
+        panorama: panorama!(a0, F, c1, N),
+        creator: DAN,
+        instance_id: 1u64,
+        value: None,
+        seq_number: 0,
+        timestamp: 52.into(),
+        round_exp: 4u8,
+        endorsed: BTreeSet::new(),
+    };
+    assert_eq!(state.validate_lnc(&d0), Some(BOB));
+    // Endorse c0.
+    d0.endorsed.insert(c0);
+    endorse!(state, rng, c0; CAROL, DAN, ALICE);
+    assert_eq!(state.validate_lnc(&d0), None);
```

Comment:
Yes, I think it's doable now.

Reply:
I would prefer to use `mut unit = WireUnit { … }` as that produces far less code.

Reply:
But wouldn't it be _much_ shorter and more readable (especially since it avoids the timestamp and round length issues which are irrelevant to this test)?

Reply:
I will make the change and you will see.

Reply:
Done in https://github.com/CasperLabs/casper-node/pull/535/commits/de7e6599a16c1ec099c31e093be5113caeca8647.